src/coreclr/gcinfo/gcinfodumper.cpp

```diff
@@ -288,15 +288,17 @@ PORTABILITY_ASSERT("GcInfoDumper::ReportPointerRecord is not implemented on this
     assert(!"unimplemented on LOONGARCH yet");
     iSPRegister = 0;
 #elif defined(TARGET_RISCV64)
-    assert(!"unimplemented on RISCV64 yet");
-    iSPRegister = 0;
+    iSPRegister = (offsetof(T_CONTEXT, Sp) - offsetof(T_CONTEXT, R0)) / sizeof(ULONGLONG);
 #endif
 
 #if defined(TARGET_ARM) || defined(TARGET_ARM64)
     BYTE* pContext = (BYTE*)&(pRD->volatileCurrContextPointers);
 #elif defined(TARGET_LOONGARCH64)
     assert(!"unimplemented on LOONGARCH yet");
     BYTE* pContext = (BYTE*)pRD->pCurrentContext;
+#elif defined(TARGET_RISCV64)
+    assert(!"unimplemented on RISCV64 yet");
+    BYTE* pContext = (BYTE*)pRD->pCurrentContext;
```

Comment:
Note, this is unimplemented block (and will not be implemented in this PR for sure). We stay with previous logic - assert in debug build and compile default code in release build. I can't remove any of this lines since this will broke previous logic - will not assert in debug or will not compile code in `#else` block.